src/main/resources/assets/paths/datasetV3ById.yaml

```diff
@@ -0,0 +1,86 @@
+  description: Updates a dataset from a valid DatasetUpdate object, returns a full dataset object
+  parameters:
+    - name: datasetId
+      in: path
+      description: The Dataset ID
+      required: true
+      schema:
+        type: integer
+  requestBody:
+    description: Returns the updated dataset. A dataset must already exist, but files can be either
+                 updated or created if they don't already exist.
+    content:
+      multipart/form-data:
+        schema:
+          type: object
+          properties:
+            dataset:
+              description:
+                type: object
+                properties:
+                  name:
+                    type: string
+                    description: The dataset name
+                  datasetName:
+                    type: string
+                    deprecated: true
+                    description: Legacy value for backwards compatibility, same as the dataset name
+                  dataSetId:
+                    type: integer
+                    format: int32
+                    description: The dataset id, integer format
+                  dacId:
+                    type: string
+                    description: The id of the DAC this dataset belongs to, if defined.
+                  datasetProperties:
+                    type: array
+                    items:
+                      $ref: './DatasetProperty.yaml'
+                    example:
+                      - propertyName: Dataset Name
+                        propertyValue: Dataset Name
+                      - propertyName: Data Type
+                        propertyValue: Whole Genome
+                      - propertyName: Species
+                        propertyValue: Human
+                      - propertyName: Phenotype/Indication
+                        propertyValue: Cardiovascular System Disease
+                      - propertyName: "# of participants"
+                        propertyValue: 20
+                      - propertyName: Description
+                        propertyValue: Description of Dataset
+                      - propertyName: dbGAP
+                        propertyValue: http://...
+                      - propertyName: Data Depositor
+                        propertyValue: NHLBI
+                      - propertyName: Principal Investigator(PI)
+                        propertyValue: NHLBI
+                      - propertyName: Sample Collection ID
+                        propertyValue: SC-000
+            alternativeDataSharingPlan:
+              description: Optional alternative data sharing plan
+              type: string
+              format: binary
+              $ref: './FileStorageObject.yaml'
+            consentGroups[0].nihInstitutionalCertificationFile:
+              description: Certification file
+              type: string
+              format: binary
+              $ref: './FileStorageObject.yaml'
```

Comment:
This is what I see when running swagger locally:
```
Resolver error at paths./api/dataset/v3/{datasetId}.put.requestBody.content.multipart/form-data.schema.properties.alternativeDataSharingPlan.$ref
Could not resolve reference: Not Found
Resolver error at paths./api/dataset/v3/{datasetId}.put.requestBody.content.multipart/form-data.schema.properties.consentGroups[0].nihInstitutionalCertificationFile.$ref
Could not resolve reference: Not Found
Resolver error at paths./api/dataset/v3/{datasetId}.put.requestBody.content.multipart/form-data.schema.properties.dataset.description.properties.datasetProperties.items.$ref
Could not resolve reference: Not Found
```
I think the syntax might be `$ref: '../schemas/FileStorageObject.yaml'` but haven't tested.

Reply:
There are a number of validation errors here.
* I don't think you can have a $ref for the two files. 
* `requestBody` can't have a description
* `properties.dataset.description` is empty
* Right below that, `type` is indented one place too far which impacts everything below it.

Reply:
This `requestBody` seems to work for me:
```
  requestBody:
    content:
      multipart/form-data:
        schema:
          type: object
          properties:
            dataset:
              description: A list of updatable dataset properties
              type: object
              properties:
                name:
                  type: string
                  description: The dataset name
                datasetName:
                  type: string
                  deprecated: true
                  description: Legacy value for backwards compatibility, same as the dataset name
                dataSetId:
                  type: integer
                  format: int32
                  description: The dataset id, integer format
                dacId:
                  type: string
                  description: The id of the DAC this dataset belongs to, if defined.
                datasetProperties:
                  type: array
                  items:
                    $ref: '../schemas/DatasetProperty.yaml'
                  example:
                    - propertyName: Dataset Name
                      propertyValue: Dataset Name
                    - propertyName: Data Type
                      propertyValue: Whole Genome
                    - propertyName: Species
                      propertyValue: Human
                    - propertyName: Phenotype/Indication
                      propertyValue: Cardiovascular System Disease
                    - propertyName: "# of participants"
                      propertyValue: 20
                    - propertyName: Description
                      propertyValue: Description of Dataset
                    - propertyName: dbGAP
                      propertyValue: http://...
                    - propertyName: Data Depositor
                      propertyValue: NHLBI
                    - propertyName: Principal Investigator(PI)
                      propertyValue: NHLBI
                    - propertyName: Sample Collection ID
                      propertyValue: SC-000
            alternativeDataSharingPlan:
              description: Optional alternative data sharing plan
              type: string
              format: binary
            consentGroups[0].nihInstitutionalCertificationFile:
              description: Certification file
              type: string
              format: binary
```

Reply:
Thank you!